tests/macsec/test_chassis_reboot.py

```diff
@@ -0,0 +1,37 @@
+'''
+This script is to test the the chassis reboot behavior for SONiC.
+
+Step 1: Configure Macsec between neighbor and DUT
+Step 2: Save config on DUT and reboot
+Step 3: Verify macsec connection is re-established after reboot
+
+'''
+
+import logging
+import pytest
+from time import sleep
+from tests.common import reboot
+from tests.common.helpers.assertions import pytest_assert
+
+logger = logging.getLogger(__name__)
+
+pytestmark = [
+    pytest.mark.topology('t2')
+]
+
+
+def test_chassis_reboot(duthost, enum_rand_one_frontend_asic_index, duthosts, localhost, enum_supervisor_dut_hostname):
+    rphost = duthosts[enum_supervisor_dut_hostname]
+    asic_index = enum_rand_one_frontend_asic_index
+    dut_lldp_table = duthost.shell("show lldp table")['stdout'].split("\n")[3].split()
+    dut_to_neigh_int = dut_lldp_table[0]
+    rphost.shell("config save -y")
+    reboot(rphost, localhost, wait=200)
+    sleep(60)  # wait extra for macsec to establish
+
+    space_var = ""
+    if asic_index is not None:
+        space_var = "-n {} ".format(asic_index)
+    macsec_status = duthost.shell("show macsec {}{}".format(space_var, dut_to_neigh_int))['stdout'].splitlines()
```

Comment:
There is no guarantee that the random asic index will refer to the ASIC corresponding to dut_to_neigh_int. How can this command call work if the asic index is referring to an ASIC that does not contain dut_to_neigh_int?

Reply:
First, what you get here, dut_to_neigh_int, is a random interface of duthost that is connected to a neighbor device, instead of a random neighbor device. I think the correct way to get this is to run command "show int status -n <asic_namespace>" and then you extract an interface out of the command output.

Reply:
I am changing this in the combined file to match what the lc reboot test does.  It will get the neighbor interface and then determine what asic to use instead of the other way its currently operating.